commands/scanandfixrepos.go

```diff
@@ -33,12 +36,22 @@ func (saf *ScanAndFixRepositories) Run(configAggregator utils.FrogbotConfigAggre
 
 func (saf *ScanAndFixRepositories) scanAndFixSingleRepository(repoConfig *utils.FrogbotRepoConfig, client vcsclient.VcsClient) error {
 	for _, branch := range repoConfig.Branches {
-		err := saf.downloadAndRunScanAndFix(repoConfig, branch, client)
+		shouldScan, checkedCommit, err := saf.shouldScanLatestCommit(context.Background(), repoConfig, client, branch)
 		if err != nil {
 			return err
 		}
+		if !shouldScan {
+			log.Info(fmt.Sprintf("Commit '%s' in repo '%s', branch '%s' has already been scanned. Skipping the scan.", checkedCommit, repoConfig.RepoName, branch))
+			continue
+		}
+		if err = saf.downloadAndRunScanAndFix(repoConfig, branch, client); err != nil {
+			err = saf.setCommitBuildStatus(client, repoConfig, vcsclient.Fail, checkedCommit, fmt.Sprintf("Frogbot error: %s", err))
+			return err
```

Comment:
Note that error we're returning here is the error that was returned from `setCommitBuildStatus` and not the error returned from `downloadAndRunScanAndFix `. Do we want this error to be returned with a nil error here?
In case we don't O suggest replacing the following code - 
```
		if err = saf.downloadAndRunScanAndFix(repoConfig, branch, client); err != nil {
			err = saf.setCommitBuildStatus(client, repoConfig, vcsclient.Fail, checkedCommit, fmt.Sprintf("Frogbot error: %s", err))
			return err
		}
```
with -
```
		if err = saf.downloadAndRunScanAndFix(repoConfig, branch, client); err != nil {
			e = saf.setCommitBuildStatus(client, repoConfig, vcsclient.Fail, checkedCommit, fmt.Sprintf("Frogbot error: %s", err))
			return errors.Join(err, e)
		}
``` 
Note that the `errors.Join` API is supported in go 1.20, and Frogbot is currently released with go 1.19.x. Let's take this opportunity and upgrade the release script of Frogbot to use go 1.20.x.